HACKING.md

```diff
@@ -65,49 +65,65 @@ Run the unit tests with:
 
     $ hack/test-go.sh
 
-or an individual package unit test with:
+or an individual package with:
 
-    $ hack/test-go.sh pkg/build
+    $ hack/test-go.sh relative/path/to/package
 
 or an individual package and all packages nested under it:
 
-    $ hack/test-go.sh pkg/build/...
+    $ hack/test-go.sh relative/path/to/package/...
 
 To run only a certain regex of tests in a package, use:
 
-    $ hack/test-go.sh pkg/build -test.run=SynchronizeBuildRunning
+    $ GOTEST_FLAGS='-test.run=SpecificTestName' hack/test-go.sh relative/path/to/package
 
-To get verbose output add `-v` to the end:
+To get verbose output for the above example:
 
-    $ hack/test-go.sh pkg/build -test.run=SynchronizeBuildRunning -v
+    $ GOTEST_FLAGS='-test.run=SpecificTestName -v' hack/test-go.sh relative/path/to/package
 
 To run all tests with verbose output:
 
-    $ hack/test-go.sh "" -v
+    $ GOTEST_FLAGS='-v' hack/test-go.sh
+
+To change the timeout for individual unit tests, which defaults to one minute, use:
+
+    $ TIMEOUT=<timeout> hack/test-go.sh
 
 To enable running the kubernetes unit tests:
 
-    $ TEST_KUBE=1 hack/test-go.sh
+    $ TEST_KUBE=true hack/test-go.sh
 
 To run unit test for an individual kubernetes package:
 
-    $ TEST_KUBE=1 hack/test-go.sh Godeps/_workspace/src/k8s.io/kubernetes/examples
+    $ hack/test-go.sh Godeps/_workspace/src/k8s.io/kubernetes/path/to/package
+
+To change the coverage mode, which is `-cover -covermode=atomic` by default, use:
 
-To turn off or change the coverage mode, which is `-cover -covermode=atomic` by default, use:
+    $ COVERAGE_SPEC="<some coverage specification>" hack/test-go.sh
 
-    $ KUBE_COVER="" hack/test-go.sh
+To turn off coverage calculation, which is on by default, use:
+
+    $ COVERAGE_SPEC= hack/test-go.sh
 
 To run tests without the go race detector, which is on by default, use:
 
-    $ KUBE_RACE="" hack/test-go.sh
+    $ DETECT_RACES= hack/test-go.sh
 
 A line coverage report is run by default when testing a single package.
 To create a coverage report for all packages:
 
-    $ OUTPUT_COVERAGE=/path/to/dir hack/test-go.sh pkg/build
+    $ COVERAGE_OUTPUT_DIR='/path/to/dir' hack/test-go.sh
 
 After that you can open `/path/to/dir/coverage.html` in the browser.
 
+To generate a jUnit XML report from the output of the tests, and see a summary of the test output 
+instead of the full test output, use:
+    
+    $ JUNIT_REPORT=true hack/test-go.sh
+
+`hack/test-go.sh` can not generate jUnit XML and a coverage report or all packages at once. If you
```

Comment:
s/can not/cannot/

s/or/for/ (unless I'm completely misparsing this sentence).